core/whistle_number_manager-1.0.0/src/wh_port_request_crawler.erl

```diff
@@ -0,0 +1,152 @@
+
+%%--------------------------------------------------------------------
+%% @doc
+%% Starts the server
+%%
+%% @spec start_link() -> {ok, Pid} | ignore | {error, Error}
+%% @end
+%%--------------------------------------------------------------------
+start_link() ->
+    gen_server:start_link(?MODULE, [], []).
+
+stop() ->
+    self() ! 'stop'.
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Initializes the server
+%%
+%% @spec init(Args) -> {ok, State} |
+%%                     {ok, State, Timeout} |
+%%                     ignore |
+%%                     {stop, Reason}
+%% @end
+%%--------------------------------------------------------------------
+init([]) ->
+    put('callid', ?MODULE),
+    lager:debug("started ~s", [?MODULE]),
+    self() ! 'start_crawling',
+    {'ok', #state{}}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling call messages
+%%
+%% @spec handle_call(Request, From, State) ->
+%%                                   {reply, Reply, State} |
+%%                                   {reply, Reply, State, Timeout} |
+%%                                   {noreply, State} |
+%%                                   {noreply, State, Timeout} |
+%%                                   {stop, Reason, Reply, State} |
+%%                                   {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+handle_call(_Request, _From, State) ->
+    {'reply', {'error', 'not_implemented'}, State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling cast messages
+%%
+%% @spec handle_cast(Msg, State) -> {noreply, State} |
+%%                                  {noreply, State, Timeout} |
+%%                                  {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+handle_cast(_Msg, State) ->
+    lager:debug("unhandled cast: ~p", [_Msg]),
+    {'noreply', State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling all non call/cast messages
+%%
+%% @spec handle_info(Info, State) -> {noreply, State} |
+%%                                   {noreply, State, Timeout} |
+%%                                   {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+handle_info('start_crawling', State) ->
+    lager:debug("crawling"),
+    Cycle = whapps_config:get_integer(?MOD_CONFIG_CAT, <<"crawler_delay_time">>, ?MILLISECONDS_IN_MINUTE),
+    wh_port_request:send_submitted_requests(),
+    erlang:send_after(Cycle, self(), 'start_crawling'),
```

Comment:
I prefer `erlang:start_timer/3`, capturing the Ref returned, and putting it in the State. Then you can `handle_info({'timeout', Ref, Message}, #state{field=Ref}=State) ->` to make sure you're handling the message you sent yourself.

If you do this, then in `init/1` you can start the timer and not message yourself to do so.